### PR TITLE
Create precise bisection tasks on completion of regression/progression.

### PR DESCRIPTION
**Pull request by oliverchang**

### CODE REVIEW

File: src/python/bot/tasks/task_creation.py [hunk 1]
```diff
@@ -16,11 +16,21 @@
...
+from build_management import revisions
+from config import local_config
 from datastore import data_handler
 from datastore import data_types
 from datastore import ndb_utils
+from google_cloud_utils import blobs
+from google_cloud_utils import pubsub
 from system import environment
 
+BISECT_SANITIZER_MAP = {
```
Comment (inferno-chromium) [inline src/python/bot/tasks/task_creation.py:28]: use environment.SANITIZER_NAME_MAP

File: src/python/bot/tasks/task_creation.py [hunk 2]
```diff
@@ -227,3 +237,57 @@ def create_tasks(testcase):
...
+  if bisect_type == 'fixed':
+    old_commit, new_commit = _get_commits(testcase.fixed, testcase.job_type)
+  elif bisect_type == 'regressed':
+    old_commit, new_commit = _get_commits(testcase.regression,
+                                          testcase.job_type)
+  else:
+    raise ValueError('Invalid bisection type ' + bisect_type)
+
+  reproducer = blobs.read_key(testcase.minimized_keys or testcase.fuzzed_keys)
+  target = testcase.get_fuzz_target()
```
Comment (inferno-chromium) [inline src/python/bot/tasks/task_creation.py:271]: We need to skip blackbox fuzzer testcases or testcase that are not fuzz target, add a bailout.

File: src/python/bot/tasks/task_creation.py [hunk 3]
```diff
@@ -227,3 +237,57 @@ def create_tasks(testcase):
...
+  if not pubsub_topic:
+    return
+
+  if bisect_type == 'fixed':
+    old_commit, new_commit = _get_commits(testcase.fixed, testcase.job_type)
+  elif bisect_type == 'regressed':
+    old_commit, new_commit = _get_commits(testcase.regression,
+                                          testcase.job_type)
+  else:
+    raise ValueError('Invalid bisection type ' + bisect_type)
```
Comment (inferno-chromium) [inline src/python/bot/tasks/task_creation.py:268]: type:

File: src/python/bot/tasks/task_creation.py [hunk 4]
```diff
@@ -227,3 +237,57 @@ def create_tasks(testcase):
   # Just create the minimize task for now. Once minimization is complete, it
   # automatically created the rest of the needed tasks.
   create_minimize_task_if_needed(testcase)
+
+
+def _get_commits(commit_range, job_type):
+  """Get commits from range."""
+  start, end = revisions.get_start_and_end_revision(commit_range)
+  components = revisions.get_component_range_list(start, end, job_type)
+  if ':' not in components[0]['link_text']:
```
Comment (inferno-chromium) [inline src/python/bot/tasks/task_creation.py:246]: assign components[0]['link_text'] to local and use in all places.

File: configs/test/project.yaml [hunk 5]
```diff
@@ -118,3 +118,6 @@ env:
   # REVISION_VARS_URL:
 
   # Any of the above variables can be overridden in a job definition.
+
+bisect_service:
```
Comment (inferno-chromium) [inline configs/test/project.yaml:122]: Should this be commented ? Will this result in error if not used or unnecessary messages accumulating.

Comment (oliverchang) [reply configs/test/project.yaml:122]: Done. Commented to be safe.

File: src/python/bot/tasks/task_creation.py (hunk 1 above)
Comment (oliverchang) [reply src/python/bot/tasks/task_creation.py]: Done! didn't realise we had this.

File: src/python/bot/tasks/task_creation.py (hunk 4 above)
Comment (oliverchang) [reply src/python/bot/tasks/task_creation.py]: done.

File: src/python/bot/tasks/task_creation.py (hunk 3 above)
Comment (oliverchang) [reply src/python/bot/tasks/task_creation.py]: done.

File: src/python/bot/tasks/task_creation.py (hunk 2 above)
Comment (oliverchang) [reply src/python/bot/tasks/task_creation.py:268]: done.

**Pull request merged**